libwild/src/layout.rs

```diff
@@ -3608,9 +3616,20 @@ fn is_symbol_undefined(
         return false;
     }
 
-    sym_file_id == sym_def_file_id
+    let is_symbol_undefined = sym_file_id == sym_def_file_id
         && symbol.is_undefined(LittleEndian)
-        && symbol_value_flags.is_absolute()
+        && symbol_value_flags.is_absolute();
+
+    match args.unresolved_symbols {
+        crate::args::UnresolvedSymbols::IgnoreAll => false,
+        crate::args::UnresolvedSymbols::ReportAll => is_symbol_undefined,
+        crate::args::UnresolvedSymbols::IgnoreInObjectFiles => {
+            is_symbol_undefined && symbol_value_flags.is_dynamic()
+        }
+        crate::args::UnresolvedSymbols::IgnoreInSharedLibs => {
+            is_symbol_undefined && !symbol_value_flags.is_dynamic()
```

Comment:
It shouldn't be necessary to check `symbol_value_flags.is_dynamic()`. The fact that we're here means that the reference is from a regular object. The `DYNAMIC` flag will often be set on undefined symbols, since if the symbol remains undefined, we can in some circumstances resolve it dynamically. See `compute_value_flags`.